services/sitemap.ts

```diff
@@ -0,0 +1,43 @@
+import { canopyManifests } from "@/services/constants/canopy";
+import { CanopyEnvironment } from "@/types/canopy";
+
+function formatDate(date) {
+  return date.toISOString().split('T')[0];
+}
+
+function buildUrl(url, domain, location, currentDate, divider) {
+  return `<url>
+    <loc>${domain}${divider}${url[location]}</loc>
+    <lastmod>${currentDate}</lastmod>
+    <changefreq>daily</changefreq>
+    <priority>0.8</priority>
+  </url>`;
+}
+
+function buildUrls(urls, domain, location, divider = '') {
+  const currentDate = formatDate(new Date());
+  return urls
+    .map((url) => buildUrl(url, domain, location, currentDate, divider))
+    .join('');
+}
+
+const generateSitemap = async () => {
+  const manifests = canopyManifests();
+  const { url, navigation } = process.env.CANOPY_CONFIG as unknown as CanopyEnvironment;
+  const currentDate = formatDate(new Date());
+  return `<?xml version="1.0" encoding="UTF-8"?>
+   <urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
+     <url>
+        <loc>${url}</loc>
+        <lastmod>${currentDate}</lastmod>
+        <changefreq>daily</changefreq>
+        <priority>0.8</priority>
+     </url>
+     ${buildUrls(navigation.primary, url, "path")}
+     ${buildUrls(navigation.about, url, "path")}
+     ${buildUrls(manifests, url, "slug", '/')}
+   </urlset>
+ `;
+};
```

Comment:
I'm not against handrolling this and it might give us more flexibility to do so. I'm wondering if we should look at using https://www.npmjs.com/package/next-sitemap as a dependency for helping us with this? It should hook the into next Next instance and build a sitemap from all static paths.